api/clients/eigenda_client.go

```diff
@@ -159,12 +158,17 @@ func (m *EigenDAClient) GetBlob(ctx context.Context, batchHeaderHash []byte, blo
 		return nil, fmt.Errorf("blob has length zero - this should not be possible")
 	}
 
-	decodedData, err := m.Codec.DecodeBlob(data)
+	encodedPayload, err := m.blobToEncodedPayload(data)
 	if err != nil {
-		return nil, fmt.Errorf("error decoding blob: %w", err)
+		return nil, fmt.Errorf("blob to encoded payload: %w", err)
 	}
 
-	return decodedData, nil
+	decodedPayload, err := codecs.DecodePayload(encodedPayload)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding payload: %w", err)
+	}
```

Comment:
Super clean! We might even want a single function like `blobToPayload` which does both operations?

Reply:
Also might want to use `blob` instead of `data` for the thing received by RetrieveBlob?

Reply:
Is a blob always power of 2?